modules/library/main/src/main/java/org/geotools/filter/function/StandardDeviationFunction.java

```diff
@@ -96,4 +110,65 @@ private Double getMax(int index, int numClasses, double average, double standard
         if (index < 0 || index >= numClasses - 1) return null;
         return Double.valueOf(average - (((numClasses / 2.0) - 1 - index) * standardDeviation));
     }
+
+    public double[] getPercentages(
```

Comment:
Should not be public. Besides that, this large block of code (this method and the ones below) seem like it could be shared with EqualIntervalFunction, or not? Guess you can do the same as with ``AbstractQuantityClassificationFunction`` and roll a common superclass.

Reply:
Thanks for your review @aaime. I've put it as private. Anyway this is not the same block of code. The EqualIntervalFunction use a GroupByVisitor to query the data and compute percentages. In this case it is creating filters mapping the breaks to query the data. Also both are different from the computePercentages method in the `AbstractQuantityClassificationFunction`

Reply:
A group by visitor seems indeed more efficient, the current one seems to be loading the entire data set from the database to run the counts? If so, that's quite inefficient.

Reply:
thanks for your point @aaime . Code has been modified so that GroupBy is used also for StandardDeviation percentages.